File: src/test/resources/samples/csharp/InheritanceSample.cs
using System;

namespace Samples.Inheritance;

public abstract class Animal
{
    public string Name { get; }
    protected Animal(string name) => Name = name;
    public virtual string Speak() => "…";
}

public class Dog : Animal
{
    public Dog(string name) : base(name) { }
    public sealed override string Speak() => "Woof";
}

public class LoudDog : Dog
{
    public LoudDog(string name) : base(name) { }
}

File: src/test/resources/samples/ruby/InheritanceSample.rb
class Animal
  attr_accessor :name
  
  def initialize(name)
    @name = name
  end
  
  def speak
    "Some sound"
  end
end

class Dog < Animal
  attr_accessor :breed
  
  def initialize(name, breed)
    super(name)
    @breed = breed
  end
  
  def speak
    "Woof!"
  end
  
  def fetch(item)
    puts "#{name} is fetching #{item}"
  end
end
